core/trietxn_test.go

```
@@ -0,0 +1,85 @@
+package core
+
+import (
+	"errors"
+	"testing"
+
+	"github.com/NethermindEth/juno/db"
+
+	"github.com/NethermindEth/juno/core/felt"
+	"github.com/bits-and-blooms/bitset"
+	"github.com/dgraph-io/badger/v3"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestTrieTxn(t *testing.T) {
+	db := db.NewTestDb()
+	prefix := []byte{37, 44}
+
+	key := bitset.New(44)
+	node := new(TrieNode)
+
+	// put a node
+	if err := db.Update(func(txn *badger.Txn) error {
+		tTxn := &TrieTxn{txn, prefix}
+
+		value, _ := new(felt.Felt).SetRandom()
+		node.UnmarshalBinary(value.Marshal())
```

Please don't ignore the error